languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -85,6 +85,61 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge @import.defs -> @name.value
 }
 
+; Import rules lifted mostly from typescript
+(program [(import_declaration)]* @import) @prog {
+  var proj_scope = ROOT_NODE
+
+  if (not (eq PROJECT_NAME "")) {
+    ; project definition
+    node proj_def__ns
+    attr (proj_def__ns) pop_symbol = "%Proj"
+    edge proj_scope -> proj_def__ns
+    ;
+    node proj_def
+    attr (proj_def) pop_symbol = PROJECT_NAME
+    edge proj_def__ns -> proj_def
+    ;
+    set proj_scope = proj_def
+  }
+
+  var mod_scope = proj_scope
+  if (not (is-empty @import)) {
+    ; module definition
+    let mod_name = (path-filestem FILE_PATH)
+    let mod_path = (path-normalize (path-join (path-dir FILE_PATH) mod_name))
+    ;
+    node mod_def__ns
+    attr (mod_def__ns) pop_symbol = "%M"
+    edge mod_scope -> mod_def__ns
+    set mod_scope = mod_def__ns
+    ;
+    scan mod_path {
+      "index$" {
+        ; skip last component for index files
+      }
+      "([^/]+)/?" {
+        node mod_def
+        attr (mod_def) pop_symbol = $1
+        edge mod_scope -> mod_def
+        set mod_scope = mod_def
+      }
+    }
+    ; make the last one a definition
+    attr (mod_scope) is_definition, source_node = @prog, empty_source_span
```

Comment:
I'd suggest doing this based on the `package` declaration, instead of the path. If you want to keep it simpler for this PR, you could decide to only support simple package names at first (with a single name like `package my_pkg`, but not with multiple components like `package a.b.c`).

Reply:
If we do it this way, am I correct in understanding that it won't allow us to support implicit package imports as the test sets up currently? Perhaps we can allow for both - use the package declaration, and if nothing is found, fall back to path? 🤔